CDP4Reporting/Parameters/ReportingParameter.cs

```diff
@@ -112,7 +118,32 @@ public ReportingParameter(string name, Type type, object defaultValue, string fi
             this.Name = name;
             this.Type = type;
             this.DefaultValue = defaultValue;
+
+            this.ForceDefaultValue = defaultValue != null && this.GetDefault(type) != defaultValue;
+
             this.FilterExpression = filterExpression;
         }
+
+        /// <summary>
+        /// Get the default value of a specific <see cref="Type"/>
+        /// </summary>
+        /// <param name="type">The <see cref="Type"/></param>
+        /// <returns>The default value</returns>
+        private object GetDefault(Type type)
+        {
+            return this.GetType()
```

Comment:
Wouldn't it be better to get rid of the helper method `GetDefaultGeeneric<T>` and just use as a body for this method (I would even inline this, since it only has one usage):
```
return type.IsValueType ? Activator.CreateInstance(type) : null;
```

Reply:
Then I would have a problem with the `bool` type:
A bool is not a value type, so it would default to null using the IsValueType/Activator.CreateInstance method.
default(bool) is false, so there is a big difference.  
I need the check to return `false`, so I'll leave it this way.